[Release-1.29] Release Notes (#17119)
istio/istio.io: content/en/news/releases/1.29.x/announcing-1.29/_index.md

We would like to thank the Release Managers for this release, **Francisco Herrera** from Red Hat, and **Petr McAllister** from Solo.io.

{{< relnote >}}

{{< tip >}}
Istio 1.29.0 is officially supported on Kubernetes versions 1.31 to 1.35.
{{< /tip >}}

## Security Update

- [CVE-2025-61732](https://github.com/advisories/GHSA-8jvr-vh7g-f8gx) (CVSS score 8.6, High): A discrepancy between how Go and C/C++ comments were parsed allowed for code smuggling into the resulting cgo binary. 
- [CVE-2025-68121](https://github.com/advisories/GHSA-h355-32pf-p2xm) (CVSS score 4.8, Moderate): A flaw in crypto/tls session resumption allows resumed handshakes to succeed when they should fail if ClientCAs or RootCAs are mutated between the initial and resumed handshake. This can occur when using `Config.Clone` with mutations or `Config.GetConfigForClient`. As a result, clients may resume sessions with unintended servers, and servers may resume sessions with unintended clients.

## What's new?

### Ambient Mesh Production-Ready Enhancements

> Istio 1.29 significantly strengthens ambient mesh capabilities with two major operational improvements enabled by default. DNS capture is now enabled by default for ambient workloads, improving security and performance while enabling advanced features like better service discovery and traffic management. This enhancement ensures that DNS traffic from ambient workloads is properly proxied through the mesh infrastructure.

Reviewer: ```suggestion
Istio 1.29 adds two operational improvements enabled by default for ambient mesh: DNS capture is now enabled by default for ambient workloads, improving security and performance while enabling advanced features like better service discovery and traffic management. This enhancement ensures that DNS traffic from ambient workloads is properly proxied through the mesh infrastructure.
```
this is nit - but seems to cut an unneeded loundness of the statement

do we use lower case for `ambient`?

Author: Yeah, I checked in previous release notes and is used ambient in lowercase

### Inference Extension Support Promoted to Beta

Support for the [Gateway API Inference Extension](https://gateway-api-inference-extension.sigs.k8s.io/) has been promoted to beta in Istio 1.29. The inference extension is an official Kubernetes project that utilizes a new `InferencePool` CRD object, along with existing Kubernetes Gateway API traffic management objects (`Gateway`, `HTTPRoute`), in order to optimize the serving of self-hosted Generative AI models in Kubernetes.

Istio 1.29 is conformant with the v1.0.1 version of the inference extension, and is available to try by enabling the `ENABLE_GATEWAY_API_INFERENCE_EXTENSION` pilot environment variable. Future releases of Gateway API Inference Extension will be supported in upcoming versions of Istio.

See [our guide](https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api-inference-extension/) and [original blog post](/blog/2025/inference-extension-support/) in order to get started.

> 

Reviewer 2: ```suggestion

### Multi-network multicluster ambient goes Beta

This release also promotes multi-network multicluster in ambient to beta status. Lots of improvements were made for robustness and completeness. The main area of focus for this transition was telemetry, where important gaps were addressed, including the implementation of more advanced peer metadata exchange in the ambient data-plane.

This means some confusing cases in multinetwork telemetry were addressed. In scenarios where Waypoints wouldn't be properly reported in L4 metrics to cases where peer information was not fully available for requests traversing different networks through an E/W Gateway.

Also, we now have [a quick guide](/docs/ambient/install/multicluster/observability) showing how to deploy Prometheus and Kiali for multi-network multicluster in ambient mode.

Note that some of these improvements may also be behind the `AMBIENT_ENABLE_BAGGAGE` feature flag mentioned in the sections above, so make sure to enable it if you want to try them out. If you need more information on how to deploy multi-network multicluster using the ambient data-plane, please follow [this guide](/docs/ambient/install/multicluster/multi-primary_multi-network/). You'll find more details about the feature on the [release notes](change-notes/).

Don't forget to share your feedback with us!

```

---
title: Announcing Istio 1.29.0
linktitle: 1.29.0
subtitle: Major Release
description: Istio 1.29 Release Announcement.
publishdate: 2026-02-12
release: 1.29.0
aliases:
    - /news/announcing-1.29
    - /news/announcing-1.29.0
---

We are pleased to announce the release of Istio 1.29. Thank you to all our contributors, testers, users and enthusiasts for helping us get the 1.29.0 release published!
We would like to thank the Release Managers for this release, **Francisco Herrera** from Red Hat, and **Petr McAllister** from Solo.io.

{{< relnote >}}

{{< tip >}}
Istio 1.29.0 is officially supported on Kubernetes versions 1.31 to 1.35.
{{< /tip >}}

## Security Update

- [CVE-2025-61732](https://github.com/advisories/GHSA-8jvr-vh7g-f8gx) (CVSS score 8.6, High): A discrepancy between how Go and C/C++ comments were parsed allowed for code smuggling into the resulting cgo binary.

> - [CVE-2025-68121](https://github.com/advisories/GHSA-h355-32pf-p2xm) (CVSS score 4.8, Moderate): A flaw in `crypto/tls` session resumption allows resumed handshakes to succeed when they should fail if ClientCAs or RootCAs are mutated between the initial and resumed handshake. This can occur when using `Config.Clone` with mutations or `Config.GetConfigForClient`. As a result, clients may resume sessions with unintended servers, and servers may resume sessions with unintended clients.

Reviewer 3: ```suggestion
```

This isn't a security release, we don't need to specify these here.

## What's new?

### Ambient Mesh Production-Ready Enhancements

Istio 1.29 adds two operational improvements enabled by default for ambient mesh: DNS capture is now enabled by default for ambient workloads, improving security and performance while enabling advanced features like better service discovery and traffic management. This enhancement ensures that DNS traffic from ambient workloads is properly proxied through the mesh infrastructure.

> Additionally, iptables reconciliation is now enabled by default, providing automatic network rule updates when the `istio-cni` DaemonSet is upgraded. This eliminates the manual intervention previously required to ensure existing ambient pods receive updated networking configuration, making ambient mesh operations more seamless and reliable for production environments.

Reviewer 3: ```suggestion
Additionally, iptables reconciliation is now enabled by default, providing automatic network rule updates when the istio-cni `DaemonSet` is upgraded. This eliminates the manual intervention previously required to ensure existing ambient pods receive updated networking configuration, making ambient mesh operations more seamless and reliable for production environments.
```
